pkg/controller/unreachable/unreachable_controller.go

```diff
@@ -159,38 +159,31 @@ func (r *ReconcileRemoteMachineSet) Reconcile(request reconcile.Request) (reconc
 	}
 
 	cdLog.Info("checking if cluster is reachable")
-	_, err = r.remoteClusterAPIClientBuilder(secretData, controllerName)
-	if err != nil {
+	status := corev1.ConditionFalse
+	reason := "ClusterReachable"
+	message := "cluster is reachable"
+	updateCheck := controllerutils.UpdateConditionNever
+	if _, err := r.remoteClusterAPIClientBuilder(secretData, controllerName); err != nil {
 		cdLog.Warn("unable to create remote API client, marking cluster unreachable")
-		cd.Status.Conditions = controllerutils.SetClusterDeploymentCondition(cd.Status.Conditions, hivev1.UnreachableCondition,
-			corev1.ConditionTrue, "ErrorConnectingToCluster", err.Error(), controllerutils.UpdateConditionAlways)
-		err := r.Status().Update(context.TODO(), cd)
-		if err != nil {
-			cdLog.WithError(err).Error("error updating cluster deployment with unreachable condition (= true)")
-			return reconcile.Result{}, err
-		}
-		cdLog.Info("cluster is unreachable")
-		return reconcile.Result{}, nil
+		status = corev1.ConditionTrue
+		reason = "ErrorConnectingToCluster"
+		message = err.Error()
+		updateCheck = controllerutils.UpdateConditionIfReasonOrMessageChange
```

Comment:
I like the code restructure but it is not related to the main fix PR is doing so should be a separate commit.

Reply:
#598